include/openmc/string_utils.h

```diff
@@ -3,6 +3,8 @@
 
 #include <string>
 
+#include "xtensor/xarray.hpp"
```

Comment:
Now that `concatenate` is a templated function, we shouldn't need this header anymore. We will need to include `<sstream>` here though. As a restult of including `<sstream>` here, we can remove the same include from `string_utils.cpp`.